plugins/linkattacher/src/linkattacherinterface.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2013-2015 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include <linkattacherinterface.h>
+
+using namespace gazebo;
+using namespace std;
+using namespace yarp::os;
+
+LinkAttacherInterface::LinkAttacherInterface() : ModelPlugin(), m_network(0), m_rpcport(0)
+{
+}
+
+LinkAttacherInterface::~LinkAttacherInterface()
+{
+  if(m_rpcport)
+  {
+    m_rpcport->close();
+    delete m_rpcport;
+  }
+
+  if(m_network)
+    delete m_network;
+}
+
+void LinkAttacherInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+{
+  if(m_network!=0)
+    return;
+
+  m_network = new yarp::os::Network();
```

Comment:
I recommend using `unique_ptr` instead of raw pointers. You don't have to remember to `delete` the object in the destructor

Reply:
@prashanthr05 if I got it correct, it recommends to use raw points and delete explicitly

Reply:
The takeaway is that, in case if you're defining a destructor for a class, it is also better to define a copy constructor and copy assignment constructor for proper resource allocation initialization.

However, if we use the smart pointers, they already do this for us. So we do not worry about cleaning up the allocated resources.